astropy/stats/sigma_clipping.py

```diff
@@ -180,6 +180,7 @@ def sigma_clipped_stats(data, mask=None, mask_val=None, sigma=3.0, iters=None):
         image.
     """
 
+    data = np.asanyarray(data)
```

Comment:
Good idea, thanks!  I changed it to handle the masking using `MaskedArray`s and 'masked_values()`.  The code is cleaner now too.